src/net.cpp

```diff
@@ -2531,6 +2521,50 @@ void CNode::EndMessage(const char* pszCommand) UNLOCK_FUNCTION(cs_vSend)
     LEAVE_CRITICAL_SECTION(cs_vSend);
 }
 
+bool CConnman::ForEachNode(std::function<bool(CNode* pnode)> func)
```

Comment:
Nit: I would rename `ForEachNode` as `ForEachNodeContinueIf` or something to signify that the return value being true is critical. I would then make a version of `ForEachNode` that does not check for the return value.

- This is more consistent* with other "foreach" API's generally, eg, boost foreach or javascript foreach iterate over all elements
 * although boost foreach does allow break
- It is unclear if `ForEachNodeContinueIf` is/will be used anywhere in the codebase, but it could introduce bugs (say, if false is somehow returned by buggy code elsewhere or later), so it's probably better to not have a check that is never used.